eng/pipelines/templates/jobs/archetype-sdk-client.yml

```
@@ -6,249 +37,195 @@ jobs:
+        build.args: '-DTRANSPORT_CURL=ON -DTRANSPORT_PAHO=ON -DAZ_PLATFORM_IMPL=WIN32 -DUNIT_TESTING=ON -DLOGGING=OFF'
         CMAKE_GENERATOR: 'Visual Studio 16 2019'
         CMAKE_GENERATOR_PLATFORM: x64
-        build.args: ' -DUNIT_TESTING=ON'
-        AZ_SDK_C_NO_SAMPLES: 'true'
-      MacOS_x64_with_unit_test:
+        PublishMapFiles: 'true'
+        MapFileArtifactSuffix: 'win-x64-dbg-prc-nolog'
+        BuildType: Debug
+
+      MacOS_Release_Preconditions_Logging_UnitTests_Samples:
+        OSVmImage: 'macOS-10.15'
+        vcpkg.deps: 'curl[ssl] paho-mqtt cmocka'
+        VCPKG_DEFAULT_TRIPLET: 'x64-osx'
+        build.args: '-DTRANSPORT_CURL=ON -DTRANSPORT_PAHO=ON -DAZ_PLATFORM_IMPL=POSIX -DUNIT_TESTING=ON'
+        PublishMapFiles: 'false'
+        BuildType: Release
+
+      MacOS_Preconditions_Logging_UnitTests:
         OSVmImage: 'macOS-10.15'
         vcpkg.deps: 'cmocka'
         VCPKG_DEFAULT_TRIPLET: 'x64-osx'
-        build.args: ' -DUNIT_TESTING=ON'
+        build.args: '-DUNIT_TESTING=ON'
         AZ_SDK_C_NO_SAMPLES: 'true'
+        PublishMapFiles: 'false'
+        BuildType: Debug
 
-      # Build with unit testing only. No samples [cmocka] - PRECONDITIONS OFF
-      # WARNINGS_AS_ERRORS          ON
-      # TRANSPORT_CURL              OFF
-      # UNIT_TESTING                ON
-      # UNIT_TESTING_MOCKS          OFF
-      # TRANSPORT_PAHO              OFF
-      # PRECONDITIONS               OFF
-      # LOGGING                     ON
-      # AZ_PLATFORM_IMPL            AZ_PLATFORM_IMPL_NONE
-      Linux_x64_with_unit_test_no_preconditions:
-        OSVmImage: 'ubuntu-18.04'
-        vcpkg.deps: 'cmocka'
+      LinuxGCC5_Release_Preconditions_UnitTests_Samples:
+        OSVmImage: 'ubuntu-16.04'
+        vcpkg.deps: 'curl[ssl] paho-mqtt cmocka'
         VCPKG_DEFAULT_TRIPLET: 'x64-linux'
-        build.args: ' -DUNIT_TESTING=ON -DPRECONDITIONS=OFF'
+        build.args: '-DTRANSPORT_CURL=ON -DTRANSPORT_PAHO=ON -DAZ_PLATFORM_IMPL=POSIX -DUNIT_TESTING=ON -DLOGGING=OFF'
+        PublishMapFiles: 'false'
+        CC: '/usr/bin/gcc-5'
         BuildType: Release
-        AZ_SDK_C_NO_SAMPLES: 'true'
-        PublishMapFiles: 'true'
-      Win_x86_with_unit_test_no_preconditions:
-        OSVmImage: 'windows-2019'
+
+      LinuxGCC5_Preconditions_UnitTests:
+        OSVmImage: 'ubuntu-16.04'
         vcpkg.deps: 'cmocka'
-        VCPKG_DEFAULT_TRIPLET: 'x86-windows-static'
-        CMAKE_GENERATOR: 'Visual Studio 16 2019'
-        CMAKE_GENERATOR_PLATFORM: Win32
-        build.args: ' -DUNIT_TESTING=ON -DPRECONDITIONS=OFF'
-        BuildType: Release
+        VCPKG_DEFAULT_TRIPLET: 'x64-linux'
+        build.args: '-DUNIT_TESTING=ON -DLOGGING=OFF'
         AZ_SDK_C_NO_SAMPLES: 'true'
-        PublishMapFiles: 'true'
-      Win_x64_with_unit_test_no_preconditions:
+        PublishMapFiles: 'false'
+        CC: '/usr/bin/gcc-5'
+        BuildType: Debug
+
+      Windows_Release:
```

Please specify x86 vs x64 in the name, for Windows:
```suggestion
      Windows_x64_Release:
```